volume/local/local_test.go

```diff
@@ -79,3 +79,46 @@ func TestInitializeWithVolumes(t *testing.T) {
 		t.Fatal("expected to re-initialize root with existing volumes")
 	}
 }
+
+func TestCreate(t *testing.T) {
+	rootDir, err := ioutil.TempDir("", "local-volume-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(rootDir)
+
+	r, err := New(rootDir, 0, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cases := map[string]bool{
+		"name":                  true,
+		"name-with-dash":        true,
+		"name_with_underscore":  true,
+		"name/with/slash":       false,
+		"name/with/../../slash": false,
+		"./name":                false,
+		"../name":               false,
+		"./":                    false,
+		"../":                   false,
+		"~":                     false,
+		".":                     false,
```

Comment:
But the test is to verify that we do indeed reject certain patterns, this should be independent of the implementation.